### DISCUSSION TITLE
소셜 서비스에 가입된 사용자 정보 획득에 대한 REST API 논의

### DISCUSSION THREAD

**Discussion by janeljs** [Sc0up/scoup-backend#53] [Refactor]
API 설계 회의 때 소셜 서비스에 가입된 사용자 정보 인증에 관련해서는 아래와 같이 논의했었는데요.  
### 소셜 인증 (소셜 계정)

- GET /api/social/authenticate
    - request
        - query param: authorization code
    - response

    ```json
    username 
    social_service_id
    social_service_type
    email
    avatar_url
    ```

PR 후 [코드 리뷰](https://github.com/Sc0up/scoup-backend/pull/22#discussion_r686515677)를 받는 과정에서 지금 설계한 path가 적합한가에 문제가 제기되었어요. 

저번 논의 때 `/social/authenticate`로 하기로 했던 이유가 scoup 서비스 측면에서 보면 권한을 부여하는 것이 아니라 인증을 하는 성격이 강하다는 의견이 있어서 였던 것 같은데, 다시 생각해보니 회원 가입도 하기 전이라서 인증이라고 보기도 애매한 것 같아요.

인증으로 생각하려면 소셜 서비스에 가입되어 있는지를 판단하는게 주된 목적이어야 할 것 같은데, 그런 목적도 있지만 주된 건 소셜 서비스에 가입된 사용자 정보를 획득하는 것 같아서요.

관련해서 `GET api/oauth/user-data` 정도로 path를 수정해보는 건 어떨까요?

**Comment by Dae-Hwa** [comment]
좋습니다.
추가적으로 회원가입 같은 api들도 너무 행동중심으로 작성했던 것 같은데,

리소스 중심으로 생각해보면 POST /users 가 되어야 적절하지 않았나 하는 생각입니다.

**Reply by janeljs** [reply]
```
POST /api/users
```
논의 결과 위의 API로 수정됨
